### DISCUSSION TITLE
Notice error escalate to exception

### DISCUSSION THREAD

**Discussion by tommy-ng0** [getsentry/sentry-php#1749] [Q&A]
Hi all, I have a weird issue. I have added PHP Sentry SDK to my project, and it is working great. Transactions, errors are showing up in our own OSS Sentry instance.

However, I noticed that any **warning** or **notice** PHP errors are now getting caught by our exception handler after initialising Sentry (the handler didn't catch these error before we added Sentry). Not sure where to look here, any ideas? Does Sentry SDK mangle `set_error_handler`?

My application is running PHP 7.4 and SDK version is 4.8.0.

```
sentry/sentry: [debug] The "Sentry\Integration\ExceptionListenerIntegration, Sentry\Integration\ErrorListenerIntegration, Sentry\Integration\FatalErrorListenerIntegration, Sentry\Integration\RequestIntegration, Sentry\Integration\TransactionIntegration, Sentry\Integration\FrameContextifierIntegration, Sentry\Integration\EnvironmentIntegration, Sentry\Integration\ModulesIntegration" integration(s) have been installed.
```

Here is an example of the error:
```
ErrorException: Notice: Undefined index:
```

**Comment by stayallive** [comment]
This is our error handler: https://github.com/getsentry/sentry-php/blob/1ea8e3d8ec549de508b3644cd09dbc251bd53b0f/src/ErrorHandler.php#L317-L355

We do convert errors to an exception to allow us to capture them. However I don't believe we leak the exception outside our own error handling and Sentry specific integration listeners. Maybe you notice something in there we mixed up?

**Reply by tommy-ng0** [reply]
Hi @stayallive, thanks for your response.

We do set our own `set_error_handler`:
```
		set_error_handler(array('handlers'), E_ERROR);
```

Interestingly, if I run this function before initialising sentry, it escalates the notice error. If I run this function after initialising Sentry, it doesn't send any notice errors anymore. If I don't run it all together, everything seems to be working as it should.
